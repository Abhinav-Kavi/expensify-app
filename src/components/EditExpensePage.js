import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense, removeExpense} from '../actions/expenses';

export class EditExpensePage extends React.Component {

  onSubmit = (expenseUpdates) => {
    const id = this.props.expense.id;
    this.props.editExpense(id, expenseUpdates);
    this.props.history.push('/');
  }

  onClick = (e)=>{
    this.props.removeExpense({id : this.props.expense.id});
    this.props.history.push('/');
  }

  render(){
    return(
      <div>
        <ExpenseForm 
        expense = {this.props.expense}
        onSubmit = {this.onSubmit}
       />
       <button 
          onClick = {this.onClick}
        >
          Remove
        </button>
    </div>
    );
  }
}

const mapStateToProps = (state,props) => ({
  expense : state.expenses.find(expense => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch,props) => ({
  editExpense : (id,expenseUpdates) => dispatch(editExpense(id,expenseUpdates)),
  removeExpense: (data) => dispatch(removeExpense(data))
});

export default  connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);