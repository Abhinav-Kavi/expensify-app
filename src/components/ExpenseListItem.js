import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';


const ExpenseListItem = ({description,amount,createdAt,id})=>(
  <div>
    <Link to={`/edit/${id}`}> <h3>{description}</h3> </Link>
    <p>
     {numeralConfig()}
     {numeral(amount).format('$0,0.00')} 
     -
     {moment(createdAt).format('MMMM Do, YYYY')}</p>   
  </div>
);

const numeralConfig = () => {
  numeral.register('locale', 'In', {
    delimiters: {
        thousands: ',',
        decimal: '.'
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
    ordinal : function (number) {
        return number === 1 ? 'er' : 'ème';
    },
    currency: {
        symbol: '₹'
    }
  });

  // switch between locales
  numeral.locale('In');
}

export default ExpenseListItem;