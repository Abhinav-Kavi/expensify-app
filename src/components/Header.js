import React from 'react';
import {NavLink} from 'react-router-dom';

const Header =()=>(
  <div>
    <h1>Expensify</h1>
    <NavLink activeClassName="is-active" to="/" exact={true}>Dashboard</NavLink>
    <NavLink activeClassName="is-active" to="/create">Create</NavLink>

  </div>
);

export default Header;