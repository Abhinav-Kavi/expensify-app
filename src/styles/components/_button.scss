.button {
  background: $blue;
  color: white;
  display: inline-block;
  border: none;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--secondary{
  background: #888;
  margin-bottom: $m-size;
}

.button--link {
  background: none;
}
