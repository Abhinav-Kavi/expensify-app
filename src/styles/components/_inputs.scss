.text-input {
  border : 1px solid #cacccd;
  font-size: $font-size-large;
  font-weight: 300;
  height: 48px;
  padding: $s-size;
}


.textarea {
  @extend .text-input;
  height: 10rem;
}


.select {
  @extend .text-input;
}
