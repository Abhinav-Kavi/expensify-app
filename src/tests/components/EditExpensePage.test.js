import React from 'react';
import {shallow} from 'enzyme';
import {EditExpensePage} from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let editExpense, removeExpense, history, wrapper;

beforeEach(()=>{
 editExpense = jest.fn();
 removeExpense = jest.fn();
 history = {push : jest.fn()};
 wrapper = shallow(
            <EditExpensePage 
              editExpense = {editExpense}
              removeExpense={removeExpense}  
              history = {history}
              expense = {expenses[0]}
            />
           );
});

test('should render EditExpensePage correctly',()=>{
  expect(wrapper).toMatchSnapshot();
});

test('should handle editExpense',()=>{
  const expenseId = expenses[0].id;
  const expenseUpdates = {
    description : expenses[0].description,
    amount : expenses[0].amount,
    createdAt : expenses[0].createdAt,
    note : expenses[0].note
  };

  wrapper.find('ExpenseForm').simulate('submit',expenseUpdates);
  expect(editExpense).toHaveBeenLastCalledWith(expenseId,expenseUpdates);
  expect(history.push).toHaveBeenLastCalledWith('/');
});

test('should handle removeExpense',()=>{
  wrapper.find('button').simulate('click');
  expect(removeExpense).toHaveBeenLastCalledWith({id : expenses[0].id});
  expect(history.push).toHaveBeenLastCalledWith('/');
});


