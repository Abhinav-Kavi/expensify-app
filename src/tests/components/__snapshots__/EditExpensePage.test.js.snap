// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 10000,
        "createdAt": -345600000,
        "description": "Credit Card Bill",
        "id": "1",
        "note": "hdfc card bill",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
